### ISSUE TITLE
Neovim 0.10.1+: Marker in wrong line when the file containing the error is opened by jump

### ISSUE THREAD

**Issue by MaxGyver83** [dhruvasagar/vim-markify#9]
Sometimes when I jump to an error in my quickfix list, the error marker is set in line 1 instead of the error line.

### How to reproduce:

`main.c`:

```c
#include "main.h"
int main(void) { return 0; }
```

`main.h`:

```c
int x = 0;
int y = y1;
```

`Makefile`:

```Makefile
main: main.o
	$(CC) $(CFLAGS) -o $@ $^
```

* Open **only** `main.c`.
* Run `:make`.
* This opens `main.h` with the cursor on `y1` (line 2) but the error marker on line 1.

![2024-11-10 18 40 27 238x89 Region](https://github.com/user-attachments/assets/ab064a90-96d8-4c99-9b0e-46bd9a36b9d2)

Running `:Markify` manually doesn't update the marker. But running `:MarkifyToggle` twice moves the marker to line 2.

Any ideas how to fix this?

**Comment by dhruvasagar** [comment]
I tested in both vim & neovim, for me the error is marked correctly on the second line always. @MaxGyver83

**Comment by MaxGyver83** [comment]
OK, thank you for trying! Then I'm going to figure out if this is related to other plugins.

**Comment by MaxGyver83** [comment]
First insight: This is only an issue (for me) in Neovim `v0.10.2` and `v0.11.0-dev-1129`. No problem in Neovim `v0.8.3`/`v0.9.5` and vim `9.1`.

**Comment by MaxGyver83** [comment]
I think this issue is not related to other plugins. I have tested it like this (using Neovim 10.2):

```sh
nvim --noplugin main.c
:packadd vim-markify
:make
```

It looks like this is an issue in Neovim 10.2 (or maybe 10.0) and newer.

**Comment by MaxGyver83** [comment]
Also tested with Neovim `v0.10.0` and `v0.10.1`:

* `v0.10.0`: No issue.
* `v0.10.1`: Same issue as in Neovim `v0.10.2`.

**Comment by dhruvasagar** [comment]
@MaxGyver83 Thanks, will try to check it out on these versions, I compile vim & neovim from source so I am using pretty recent versions of both

**Comment by MaxGyver83** [comment]
@dhruvasagar : Have you already updated your Neovim to version 0.10.1 (or newer)?

**Comment by dhruvasagar** [comment]
@MaxGyver83 Yes, I build neovim from source and I update it reasonably frequently.

**Comment by dhruvasagar** [comment]
```
NVIM v0.11.0-dev-1421+g6d2c67350a
Build type: Release
LuaJIT 2.1.1734355927
Run "nvim -V1 -v" for more info
```

**Comment by MaxGyver83** [comment]
Can you reproduce this issue? I have just tried _Neovim v0.11.0-dev-1454+gefe1732c6f_ and I still experience it.

**Comment by dhruvasagar** [comment]
@MaxGyver83 no I am not seeing this issue.

**Comment by MaxGyver83** [comment]
OK, thank you for checking. Then it must be related to my `.vimrc`. I have tried to test it in a clean environment. Maybe the environment wasn't clean enough.

**Comment by MaxGyver83** [comment]
How do you test in a clean environment?

I have tried this:

```sh
nvim-v0.11.0-dev-1454+gefe1732c6f.appimage --clean main.c
```

```vim
:set signcolumn=yes
:set packpath^=~/.vim/pack/plugins
:packadd vim-markify
```

But I think `vim-markify` wasn't loaded. When I type `:Ma<tab>`, I don't get any `Markify` commands suggested.

**Comment by dhruvasagar** [comment]
currently I have tested in my neovim / vim setup, but it should be easy to setup a minimal vimrc that sets up vim-markify. If you aren't getting `:Ma<tab>` then something is off. Will try it out.

**Comment by MaxGyver83** [comment]
I can reproduce it like this, with Neovim 0.10.3:

```sh
$ nvim -u NONE main.c
```

```vim
:set runtimepath+=/home/max/.vim/pack/plugins/opt
:runtime vim-markify/plugin/markify.vim
:make
```

Same result for:

```sh
nvim-v0.11.0-dev-1454+gefe1732c6f.appimage -u NONE main.c
```

**Comment by dhruvasagar** [comment]
@MaxGyver83 I was able to reproduce this. Feels like the root cause is that the error list, when you look at getqflist(), it includes a bunch of errors. Will dig deeper.
